packages/components/src/custom-select-control-v2/stories/index.story.tsx

```diff
@@ -25,6 +25,7 @@ const meta: Meta< typeof CustomSelect > = {
 		renderSelectedValue: { control: { type: null } },
 		value: { control: { type: null } },
 	},
+	tags: [ 'sb-wip' ],
 	parameters: {
 		badges: [ 'wip' ],
```

Comment:
I wonder if there's a way to WIP or private statuses using only one property in the story's meta, instead of having to specify it twice (`tags` and `parameters.badges`).

I don't personally know much about Storybook config, but @mirka may know a way

Reply:
> I wonder if there's a way to WIP or private statuses using only one property in the story's meta...

I've not seen one. Add-ons (like [the badge plugin we're using](https://storybook.js.org/addons/@geometricpanda/storybook-addon-badges)) take their data from `parameters`, which aren't loaded until the page is rendered for the first time. `tags` was the only property I could find that was available globally from the start.

But yes, removing the duplication would be ideal, and the long-term goal – the intent of the PR was to get _something_ working, before spending the time simplifying.

In my head, I was thinking we could do something like this (with a little bit of TypeScript magic to maintain the associations)...

```ts
function MetaProcessor( meta ) {
  const { tags = [], parameters = {} } = meta;
  const { badges = [] } = parameters;
  const badgeTags = badges.map( ( badge ) => `sb-${ badge }` );

  return { ...meta, tags: tags.concat( badgeTags ) };
}
```

```diff
- const meta: Meta< typeof Component > = {
+ const meta = MetaProcessor( {
    ...
    parameters: {
      badges: [ 'private', 'wip' ],
      ...
    },
    ...
- };
+ } );
```

Ultimately it would be nice if we could deduce any badges from the code, but that's very much not an immediate need!

Reply:
And by "TypeScript magic", I meant this kind of thing...

```ts
type Component = Meta[ 'component' ];
type Badges = {
  parameters?: Meta[ 'parameters' ] & {
    badges?: string[];
  };
};
type MetaProp< C extends Component >
  = Meta< C >
  & { component?: C; }
  & Badges;

function MetaProcessor< C extends Component >(
  meta: MetaProp< C >
): Meta< C > {
  ...
}

...

// implicitly... meta: Meta<Component>
const meta = MetaProcessor( {
  component: Component,
  ...
} );
```